channels/channel.videoland/videolandnl/chn_videolandnl.py

```diff
@@ -255,6 +257,34 @@ def extract_program_id(self, data: str) -> Tuple[Union[str, JsonHelper], List[Me
         self.__program_id = json_data.get_value("entity", "id", fallback=None)
         return json_data, []
 
+    def postprocess_episodes(self, data, items: list[MediaItem]) -> list[MediaItem]:        
+        # Season episodes can be named ["Aflevering 1", "Aflevering 2"] and so on, or ["Some episode name", "Another episode name", "Yet another episode name"]
+        # without an episode number. Some episodes have a date, and some have no date, even within a single season.
+
+        # All this messes up Kodi episode ordering by either date or name:
+        # If no episodes have a date, Kodi orders by name, so "Another episode name" becomes the first episode instead of the second.
+        # If some episodes have a date and some later ones don't, Kodi shows the later ones as the first episodes (mindate), followed by the ones with a date.
+
+        # This postprocessing function fixes name ordering by adding an episode number in front of the title if the episodes
+        # are not like "Aflevering 1", "Aflevering 2" and so on.
+        # Episode names become ["01 Some episode name", "02 Another episode name", "03 Yet another episode name"].
+        # Numbered episodes are fairly common, and in this case it is useful to achieve the right sort order by name.
+
+        # This postprocessing function fixes date ordering by setting the date of an episode, if it has no date, to the date of the previous episode.
+        if (len(items) > 1
+            and data.json.get("featureId") == "videos_by_season_by_program"
+            and all(item.media_type == "episode" and not re.match("[Aa]flevering \d+", item.name) for item in items)):
+            date = ""
+            for index, item in enumerate(items, start=1):
+                item.name = f"{index:02} {item.name}"
+                if (not item.has_date() and date):
+                    timestamp = datetime.datetime.strptime(date, "%Y-%m-%d")
```

Comment:
Could you use `DateHelper.get_datetime_from_string`?

Reply:
Sure!